Update Mini-Cart block logic to use isSiteEditorPage util

Aljullu opened the pull request:
In #9468, @gigitux introduced a `isSiteEditorPage()` util to detect when we are in the site editor.

This PR reverts https://github.com/woocommerce/woocommerce-blocks/pull/9442 and updates the Mini-Cart block logic code to use that util, instead of its own solution.

### Testing

#### User Facing Testing

1. Create a post or page.
2. Add the Mini Cart block and select it.
3. Verify in the sidebar there is no _Mini-Cart in cart and checkout pages_ toggle.
4. Now go to the Site Editor (Appearance > Editor).
5. Add the Mini Cart block to the header of your theme.
6. Verify in the sidebar there is a _Mini-Cart in cart and checkout pages_ toggle.
![imatge](https://github.com/woocommerce/woocommerce-blocks/assets/3616980/96bccc7b-3a1d-479e-a98b-54b600582747)
7. Now change to a classic theme (ie: Storefront).
8. Go to Appearance > Widgets and add the Mini Cart block to a widget area.
9. Verify in the sidebar there is no _Mini-Cart in cart and checkout pages_ toggle.

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

Aljullu committed:
Revert "Mini-Cart: prevent 'Mini-Cart in cart and checkout pages' toggle from showing up in the post/page editor (#9442)"

This reverts commit 31050700cd2f93e432c32791382de549cf2160bd.

Aljullu committed:
Update Mini-Cart block logic to use isSiteEditorPage util

kmanijak reviewed (commented):
I tested this and:
- ✅ **There is no** _Mini-Cart in cart and checkout pages_ toggle in post/page editor
- ✅ **There is** _Mini-Cart in cart and checkout pages_ toggle in site editor
- ❌ **There is no** _Mini-Cart in cart and checkout pages_ toggle when added in Appearance -> Widget of classic theme

<img width="1300" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/cb93f70d-c027-42ab-8809-4083adcd28a5">

Let me know if I may be doing something wrong 🤔

gigitux commented:
[Reading the code](https://github.com/woocommerce/woocommerce-blocks/pull/9501/files#diff-b4fda8cb4e251e52815013405eedcbde599070246e222f8cc0576d41d7972db9R77-R108), the check works well. If you read it, the `Mini-Cart in the cart and checkout pages option` should be visible only in the Site Editor.

The testing instructions aren't correct, or it is necessary to add another check like "isWidgetArea".

Aljullu commented:
> x There is no Mini-Cart in cart and checkout pages toggle when added in Appearance -> Widget of classic theme

Ouch, I don't know what I did wrong when testing. :sweat_smile:  Probably I didn't build properly. Sorry about the confusion.

I updated the testing steps accordingly. While I think there is value in making that feature available in the _Widgets_ screen, we haven't received any reports asking for it, and it's not sure if we are going to keep it in the future, so maybe better not to enable it in more places for now. Until #9442 that feature wasn't enabled in the _Widgets_ screen either, so we would be keeping things as they were.

@kmanijak I asked for a re-review, but I didn't do any code changes, I just updated the testing steps.

kmanijak reviewed (approved):
> @kmanijak I asked for a re-review, but I didn't do any code changes, I just updated the testing steps.

✅  There is no Mini-Cart in cart and checkout pages toggle when added in Appearance -> Widget of classic theme

All's good then! Approving! 💪
